### DISCUSSION TITLE
Any plans to add conversation history support?

### DISCUSSION THREAD

**Discussion by Stevenic** [microsoft/TypeChat#72] [General]
These models all benefit from seeing themselves follow instruction so you really need to feedback to them their chat history with the user. `gpt-3.5-turbo` in particular, benefits from including a 1 shot example of it returning the JSON you expect back. Even if you don't include a 1 shot example you'll find that the more you show the model returning the responses you expect the more reliable its output will be. The [current prompt](https://github.com/microsoft/TypeChat/blob/152b2d35e0fc81935a442bc04e85af0dd99669b0/src/typechat.ts#L79) doesn't look to have any provisions for sending the model conversation history so wondering if that's been thought about?

Additionally, all of the newer OpenAI models use the Chat Completion APIs which are message based so you really want to pass that conversation history as a message array.

**Comment by steveluc** [comment]
Thanks!  We are aware that judicious use of examples increases accuracy.  There are some examples in the TypeChat code, such as in this commented api from the music application.  
```
    // return the last track list shown to the user
    // for example, if the user types "play the third one" the player plays the third track
    // from the last track list shown
    getLastTrackList(): TrackList;
```

Like with calibrating the amount and directness of repair instructions, we are taking a minimalist approach at this point, seeing how far we can go with minimal repair, instruction, and examples.  This gives us head room as we take on larger applications, so that we can tune up these additional filters in a way that aligns with concrete scenarios.

**Comment by steveluc** [comment]
on conversation history see #39
